x-pack/solutions/observability/plugins/observability/public/pages/alert_details/hooks/related_alerts/use_build_related_alerts_query.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import {
+  ALERT_END,
+  ALERT_GROUP,
+  ALERT_INSTANCE_ID,
+  ALERT_RULE_TAGS,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import dedent from 'dedent';
+import moment from 'moment';
+import { ObservabilityFields } from '../../../../../common/utils/alerting/types';
+import { TopAlert } from '../../../../typings/alerts';
+
+interface Props {
+  alert: TopAlert<ObservabilityFields>;
+}
+
+export function useBuildRelatedAlertsQuery({ alert }: Props): QueryDslQueryContainer {
+  const groups = alert.fields[ALERT_GROUP];
+  const shouldGroups: QueryDslQueryContainer[] = [];
+  groups?.forEach(({ field, value }) => {
+    if (!field || !value) return;
+    shouldGroups.push({
+      bool: {
+        boost: 2.0,
+        must: [
+          { term: { 'kibana.alert.group.field': field } },
```

Comment:
nit: we can use `[ALERT_GROUP_FIELD]` and `[ALERT_GROUP_VALUE]` in place of `kibana.alert.group.field` and `kibana.alert.group.value`